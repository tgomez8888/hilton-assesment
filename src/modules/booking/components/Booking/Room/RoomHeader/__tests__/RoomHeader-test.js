import { mount } from "enzyme";
import {RoomHeader} from "..";
import { testComponent } from "../../../../../../../utils/test/testComponent";

describe("RoomHeader", () => {
  function render({ room, onChange }) {
    return mount(testComponent(RoomHeader, {room, onRoomValuesChange: onChange}));
  }

  test("renders only label when room priority is 1", () => {
    const roomPriority1 = { id: 1, priority: 1, selected: true };
    const wrapper = render({ room: roomPriority1, onChange: jest.fn() });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("renders checkbox label when room priority greater than 1", () => {
    const roomPriority2 = { id: 5, priority: 2, selected: false };
    const wrapper = render({ room: roomPriority2, onChange: jest.fn() });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("onChange is called when checkbox clicked", () => {
    const testFunction = jest.fn();
    const onChange = e => testFunction(e.target.name, e.target.value);

    const wrapper = render({
      room: { id: 5, priority: 2, selected: false},
      onChange
    });
    wrapper
      .find("input")
      .first()
      .simulate("change", { target: { name: "selected", value: true } });

    expect(testFunction).toBeCalledWith("selected", true);
  });
});
