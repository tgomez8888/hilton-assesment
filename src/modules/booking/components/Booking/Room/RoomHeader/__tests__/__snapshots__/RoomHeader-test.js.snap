// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomHeader renders checkbox label when room priority greater than 1 1`] = `"<div class=\\"card-header\\"><input type=\\"checkbox\\" name=\\"selected\\"/>Room 5</div>"`;

exports[`RoomHeader renders only label when room priority is 1 1`] = `"<div class=\\"card-header\\">Room 1</div>"`;
